### ISSUE TITLE
[Feature Request]: Support OAuth Authentication for Salesforce Input/Output

### ISSUE THREAD

**Issue by zoomingrocket** [apache/hop#4657]
### Feature Request: OAuth Authentication Support for Salesforce Integration

**Summary:**
To enhance security and comply with evolving Salesforce policies, we propose adding OAuth authentication support for Salesforce Input/Output operations in our product.

**Background:**
Many companies are adopting stricter security measures for their Salesforce Organizations, requiring all API access to be conducted via a Connected App using OAuth. This policy renders the standard SOAP API login() call with username, password, and security token ineffective. Instead, an OAuth handshake must be performed to generate a bearer token, which is then used in the SOAP "sessionID" header for all Salesforce SOAP calls.

**Proposed Solution:**
1. **OAuth Authentication Mechanism:**
   - Implement OAuth as an authentication option within Salesforce Input/Output operations.
   - Allow users to perform the OAuth handshake and generate a bearer token.

2. **REST Client Integration:**
   - Enable the REST Client to generate an OAuth bearer token.
   - Allow the insertion of the "sessionID" SOAP header parameter for Salesforce Input/Output operations without requiring additional username/password setup.

**References:**
- Salesforce OAuth and SOAP API: https://developer.salesforce.com/blogs/developer-relations/2011/03/oauth-and-the-soap-api
- Other ETL Vendors Reference: https://knowledge.informatica.com/s/article/515185?language=en_US
- Salesforce Security Policy: https://help.salesforce.com/s/articleView?id=sf.security_api_access_control_all_users.htm&type=5

This feature will align our product with current security practices and ensure seamless integration with Salesforce for our users.

### Issue Priority

Priority: 2

### Issue Component

Component: Hop Gui

**Comment by bamaer** [comment]
Absolutely. A first stab at that will be included in 2.11, but there are a lot of possible options and configurations. We'll need to work on this for a couple of releases.

**Comment by zoomingrocket** [comment]
@CatalyseRichard  @bamaer  - A dedicated OAuth Token generator will help, but the current problem is passing the generated bearer token to Salesforce SOAP (WSDL) based API, which requires the sessionID header.

**Comment by bamaer** [comment]
.take-issue
